code/modules/mechs/mech_interaction.dm

```diff
@@ -1,4 +1,8 @@
 /mob/living/exosuit/MouseDrop_T(atom/dropping, mob/user)
+	if (user.buckled)
```

Comment:
Put it in check_enter instead, which is somewhere around line 168. That's where enter checks should be.

It'll prevent people from attempting to enter, starting the do_after, then buckling themselves and avoiding this check.

Reply:
Good idea, I didn't even notice that section. Updated the pull.